core/store/src/lib.rs

```diff
@@ -67,8 +69,9 @@ impl Store {
     /// Caller must hold the temporary directory returned as first element of
     /// the tuple while the store is open.
     pub fn tmp_opener() -> (tempfile::TempDir, StoreOpener<'static>) {
+        static CONFIG: Lazy<StoreConfig> = Lazy::new(StoreConfig::test_config);
```

Comment:
I am not sure I understand why we need to cache the config here.  Can we not always call `test_config`?  Is calling `test_config` a very expensive operation or can it return different values when called again and again?

Reply:
cc @mina86, this is the kind of side-effects which made me to avoid lifetime parameters by default (echoing back https://github.com/near/nearcore/pull/6973#pullrequestreview-996449563).

Reply:
This is one of changes I took from https://github.com/near/nearcore/pull/7027, and I don't fully understand the reasoning as well. Is it fine to proceed with this PR and discuss this moment in #7027?